### PR TITLE
Issue 22076 - Fix regressions in dupctx testing

### PR DESCRIPTION
**Pull request by nhorman**
Hit two regressions in dupctx tests from my addition of those methods in the various ciphers.  Fix them up

##### Checklist

- [x] tests are added or updated

### CODE REVIEW

File: providers/implementations/ciphers/cipher_aes_gcm.c [hunk 1]
```diff
@@ -37,10 +37,14 @@ static void *aes_gcm_newctx(void *provctx, size_t keybits)
 static void *aes_gcm_dupctx(void *provctx)
 {
     PROV_AES_GCM_CTX *ctx = provctx;
+    PROV_AES_GCM_CTX *dctx = NULL;
 
     if (ctx == NULL)
         return NULL;
-    return OPENSSL_memdup(ctx, sizeof(*ctx));
+    dctx = OPENSSL_memdup(ctx, sizeof(*ctx));
+    if (dctx != NULL && dctx->base.gcm.key) {
```
Comment (paulidale) [inline providers/implementations/ciphers/cipher_aes_gcm.c:45]: `!= NULL` after key.

**Review by paulidale — APPROVED** [review]
With the style nit fixed.

**Comment by t8m** [comment]
SM4-CCM needs a fix too.

**Comment by t8m** [comment]
And SM4-GCM and ARIA-GCM

**Comment by t8m** [comment]
SM4-GCM still needs the treatment.

**Comment by t8m** [comment]
Now this looks promising. If you agree, please make it ready for review.

**Comment by nhorman** [comment]
Ok, all checks are passing now, and I've run the fips cross-version check here manually.  with the dupctx commits restored, plus these two commits, running openssl 3.0 with a 3.1 fips provider passes as well

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
For 3.0 and 3.1 branches we will need separate PRs because this won't cherry pick cleanly and both these branches will need also the changes from:
#21933 and #22094

There is no rush in doing the PRs as they should be merged only after the 3.1 and 3.0 releases on Tuesday anyway.

**Comment by nhorman** [comment]
@paulidale style nits fixed, if you could please review/approve

**Comment by t8m** [comment]
@nhorman I assume you've misclicked. As the github.com repo is not the primary one, this is not an issue. The commit is already removed.

Could you please create a new PR from this branch?

**Comment by nhorman** [comment]
@t8m I can, but I'm not sure what you mean when you say the github repo is not the primary one.  what repo are you asking me to target here?

**Comment by paulidale** [comment]
The repo on GitHub is a copy of a [repo in our internal git](https://github.openssl.org/openssl/openssl).
Changes pushed here will be overwritten by the next change to the primary repo.

Most of us use the `ghmerge` script from the [tools repository](https://github.com/openssl/tools).
You need committer permissions to push changes to the primary repository (which you don't have yet).

**Comment by t8m** [comment]
> @paulidale ok, thank you for that. So what's the process for me then, as someone who doesn't have commit access? Shall I just leave prs on the public repo open until someone merges them? I don't see any process documented in @hlandau 's new developer document for this situation.

Yes, you should leave the PR open and let someone from @openssl/committers merge it.

However since GH already thinks this PR was merged (although in reality it was not), you must open a new PR here with the same changes.

**Comment by paulidale** [comment]
We could just cherry pick the two commits to master with the addition of the approved by lines in the commit?

**Comment by nhorman** [comment]
https://github.com/openssl/openssl/pull/22102

**Pull request merged**